apps/daimo-mobile/src/logic/trpc.tsx

```diff
@@ -56,69 +66,118 @@ function chooseChain<T>({
   else return testnet;
 }
 
-function getOpts(daimoChain: DaimoChain) {
-  return {
-    links: [
-      httpBatchLink({
-        url: chooseChain({
-          daimoChain,
-          mainnet: apiUrlMainnetWithChain,
-          testnet: apiUrlTestnetWithChain,
+const customTRPCfetch = async (
+  input: RequestInfo | URL,
+  init?: RequestInit
+) => {
+  const url = (() => {
+    if (input instanceof URL) return input;
+    else if (input instanceof Request) return new URL(input.url);
+    else return new URL(input);
+  })();
+
+  init = init ?? {};
+  init.headers = (init.headers ?? {}) as Record<string, string>;
+
+  const platform = `${Platform.OS} ${Platform.Version}`;
+  const version = `${nativeApplicationVersion} #${nativeBuildVersion}`;
+  init.headers["x-daimo-platform"] = platform;
+  init.headers["x-daimo-version"] = version;
+
+  // Fetch timeout
+  const { pathname } = url;
+  const func = pathname.split("/").slice(-1)[0] as keyof AppRouter;
+  const timeout = (() => {
+    if (func === "deployWallet") return 60_000; // 1 minute
+    else return 10_000; // default: 10 seconds
+  })();
+  console.log(`[TRPC] fetching ${url}, timeout ${timeout}ms`, init);
+  const controller = new AbortController();
+  const timeoutID = setTimeout(() => {
+    console.log(`[TRPC] timeout after ${timeout}ms: ${input}`);
```

Comment:
i think this will always print `[TRPC] timeout ...`, even if the function didn't time out

Reply:
fixed